### PR TITLE
Update solution to VS2017

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj [hunk 1]
```diff
@@ -1,128 +1,46 @@
...
-    <Compile Include="**\*.cs" Exclude="**\obj\**\*;**\*.Designer.cs;Protobuf\AnalyzerReport.cs" />
-    <Compile Include="Protobuf\AnalyzerReport.cs">
-      <!-- Exclude the file from analysis -->
-      <SonarQubeExclude>true</SonarQubeExclude>
-    </Compile>
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.SonarAnalyzer.Common.config" />
-    <None Include="Protobuf\AnalyzerReport.proto">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj]: This file now appears in VS2017 "_Copy to output=Do not copy_". However, the build still works and the _AnalyzerReport.cs_ is still regenerated correctly when the _AnalyzerReport.proto_ file changes, so perhaps it wasn't required.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj [hunk 2]
```diff
@@ -1,128 +1,46 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Google.Protobuf" Version="3.1.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="1.0.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.1.36" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Remove="Protobuf\AnalyzerReport.cs" />
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SonarAnalyzer.Common.csproj]: There's a new _Update_ attribute that can be used, specifically to allow modification of metadat for files that were included by globbing: https://docs.microsoft.com/en-us/visualstudio/msbuild/item-element-msbuild

File: sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/SonarAnalyzer.Utilities.csproj [hunk 3]
```diff
@@ -1,140 +1,46 @@
...
-    <Optimize>true</Optimize>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
     <OutputPath>bin\Classic\Release\</OutputPath>
-	<IntermediateOutputPath>obj\Classic</IntermediateOutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <AppConfig>app.classic.config</AppConfig>
+    <IntermediateOutputPath>obj\VS2Classic015</IntermediateOutputPath>
```
Comment (duncanp-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/SonarAnalyzer.Utilities.csproj]: Error in folder name, but probably doesn't matter in practice.

File: sonaranalyzer-dotnet/SonarAnalyzer.sln [hunk 4]
```diff
@@ -1,82 +1,85 @@
...
-		HideSolutionNode = FALSE
-	EndGlobalSection
-	GlobalSection(NestedProjects) = preSolution
-		{4AC7D054-B645-46D7-9671-D666BCE0C745} = {B7233F78-E142-4882-B084-7C83BE472109}
-		{E4D5FF22-C1A4-4CAD-B068-E47817F25848} = {F51A8C2A-A0F5-4282-956F-549D05860396}
-		{98EF48F7-28D7-44CA-8C7F-6042EA4898E1} = {F51A8C2A-A0F5-4282-956F-549D05860396}
-	EndGlobalSection
-EndGlobal
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
[the remark this answers is not in the document]
Comment (michalb-sonar) [reply sonaranalyzer-dotnet/SonarAnalyzer.sln]: I'd keep VS2017 in case we need it for C#7.

**Pull request merged**